ethereum/contracts/BasicInboundChannel.sol

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.5;
+pragma experimental ABIEncoderV2;
+
+import "./ParachainLightClient.sol";
+import "./BeefyLightClient.sol";
+
+contract BasicInboundChannel {
+    uint256 public constant MAX_GAS_PER_MESSAGE = 100000;
+    uint256 public constant GAS_BUFFER = 60000;
+
+    uint64 public nonce;
+
+    BeefyLightClient public beefyLightClient;
+
+    struct Message {
+        address target;
+        uint64 nonce; // TODO: this might cause an error, we use uint256 when encoding on Parachain
```

Comment:
yeh, can you confirm the type/fix? we just need to change this to a uint256 or?

Reply:
uint64 is the correct type for message nonces in both Substrate and Solidity.

The ABI-encoder in Substrate is doing the correct encoding for a uint64, since the output encoding of a uint64 and uint256 are exactly the same (32 bytes wide). The encoder library is just very low-level and so just doesn't provide convenience functions to directly encode a uint64.

Reply:
so should probably remove the TODO as well.

Reply:
Removed in 8848a4f7b396ea712e8d619034b3ab4795a85100